### ISSUE TITLE
Catch com.epicgames.launcher links on windows

### ISSUE THREAD

**Issue by NoelJacob** [derrod/legendary#644]
It shows cannot find app on microsoft store

**Comment by derrod** [comment]
Fill in the template to explain what you mean or this will be closed.

**Comment by RehanPlayz** [comment]
Having the same issue when trying to launch GTA V (Premium Edition) from Epic Games
![image](https://github.com/derrod/legendary/assets/63925635/45aec4c6-0b36-453f-9616-d0f641003784)

**Comment by RehanPlayz** [comment]
> Having the same issue when trying to launch GTA V (Premium Edition) from Epic Games ![image](https://private-user-images.githubusercontent.com/63925635/307536667-45aec4c6-0b36-453f-9616-d0f641003784.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg3OTMzNjksIm5iZiI6MTcwODc5MzA2OSwicGF0aCI6Ii82MzkyNTYzNS8zMDc1MzY2NjctNDVhZWM0YzYtMGIzNi00NTNmLTk2MTYtZDBmNjQxMDAzNzg0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI0VDE2NDQyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM1NGMwZGU2YTQ4YmUwZWU0NzVhMzNmNjA2NmU0ZTg5OGUyOWM0YWVlY2QzYjgzZDVkMTVhZjFhMDEyOWMyZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.arTvS_pCS3PGS6x_p9vuz5rQaNT74x0_9p7NrF57cTg)

Found a fix for this on the Heroic Game launcher discord, you apparently need to start the Rockstar launcher and then start the game with the launcher already running.

The user queenbiscuit311 posted this batch script to automate it over there.
```
start /B "null" "C:\Program Files\Rockstar Games\Launcher\LauncherPatcher.exe"

ping -n 10 localhost > nul

start heroic://launch/legendary/9d2d0eb64d5c44529cece33fe2a46482
```

**Comment by Karl255** [comment]
Can something be done about this? I have the same issue when trying to launch GTA V. The fix mentioned above doesn't do anything for me (I didn't even use Heroic before this and installing it changed nothing). The Rockstar Games Launcher is running, so this isn't the problem. I can't start the game at all like this.

**Comment by anadius** [comment]
Open a terminal as an administrator and paste this: `reg add HKCR\com.epicgames.launcher` (alternatively you can open the registry and add `com.epicgames.launcher` key yourself to `HKEY_CLASSES_ROOT`).

This is what helped for me. Then I started the game with `legendary launch 9d2d0eb64d5c44529cece33fe2a46482` and the Rockstar Games Launcher asked me to link my accounts. And in fact it DIDN'T work when I ran the `legendary` command with the launcher open, I had to close it first.

So in short, add the registry key, close everything, run the game with `legendary launch <game ID>`

**Comment by anadius** [comment]
Had the game installed for all this time but haven't played. It worked then, it doesn't work now. Rockstar Launcher opens and shows a "Buy now" button. I've tried messing with the registry some more, checking what keys the launcher reads with Process Monitor, but I have no clue what's wrong. It tries to relaunch the game. But when I set `HKCR\com.epicgames.launcher\shell\open\command` to point to legendary (with command line args to run GTA 5) it's stuck in a loop. Maybe Epic launcher opens the game with some different arguments, I don't know.

**Comment by Zharevnart** [comment]
> Rockstar Launcher now checks if the parent process is called `EpicGamesLauncher.exe` and if it's still running. There's a wrapper available [here](https://github.com/Etaash-mathamsetty/heroic-epic-integration). Download it, rename to `EpicGamesLauncher.exe` and put in your game installation folder.
> 
> Then you run the game with `legendary launch <game ID> --wrapper EpicGamesLauncher.exe`
> 
> Registry key is no longer needed.

update 2026 place EpicGamesLauncher.exe beside legendary.exe 
Then you run the game with `legendary launch <game ID> --wrapper EpicGamesLauncher.exe`

**Comment by Zharevnart** [comment]
this fix for epic games launcher if you're guys still opened epic
https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/wiki/Rockstar-Games-from-Epic-Games
